### PR TITLE
Improve error messages when limits reached

### PR DESCRIPTION
**Pull request by kirill-sizov**
### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into [CHANGELOG](https://github.com/cvat-ai/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/cvat-ai/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

**Comment by nmanovic** [comment]
@kirill-sizov , is the PR ready for merge?

**Comment by kirill-sizov** [comment]
> @kirill-sizov , is the PR ready for merge?

Not yet

File: cvat/apps/iam/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from rest_framework.exceptions import PermissionDenied
+
+class LimitsReachedError(PermissionDenied):
+    default_personal_detail = "You've reached the maximum number of {}. Contact the administrator to extend the limits."
+    default_org_detail = "You've reached the maximum number of {}. Contact the administrator to extend the limits for organization."
```
Comment (nmanovic) [inline cvat/apps/iam/exceptions.py:5]: Probably we can add here the slug for an organization

 ```suggestion
    default_org_detail = "You've reached the maximum number of {}. Contact the administrator to extend the limits for `{}` organization."
```

File: cvat/apps/iam/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from rest_framework.exceptions import PermissionDenied
```
Comment (nmanovic) [inline cvat/apps/iam/exceptions.py:1]: Need to add copyright header for all new files.

File: cvat/apps/engine/errors.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+from django.conf import settings
```
Comment (nmanovic) [inline cvat/apps/engine/errors.py:1]: Need to add the copyright header

Comment (nmanovic) [reply cvat/apps/engine/errors.py:5]: Can we merge iam/exceptions.py and engine/errors.py? I would remove the file.

Comment (kirill-sizov) [reply cvat/apps/engine/errors.py:5]: Done

File: tests/python/rest_api/test_limits.py [hunk 4]
```diff
@@ -391,10 +387,10 @@ def setup(
     _DEFAULT_PROJECTS_LIMIT = 3
     _DEFAULT_CLOUD_STORAGES_LIMIT = 10
 
-    _TASK_LIMIT_MESSAGE = "org tasks limit reached"
-    _PROJECT_TASK_LIMIT_MESSAGE = "org project tasks limit reached"
-    _PROJECTS_LIMIT_MESSAGE = "org projects limit reached"
-    _CLOUD_STORAGES_LIMIT_MESSAGE = "org cloud storages limit reached"
+    _TASK_LIMIT_MESSAGE = "tasks per organization"
+    _PROJECT_TASK_LIMIT_MESSAGE = "tasks per organization project"
```
Comment (mdacoca) [inline tests/python/rest_api/test_limits.py:391]:
```suggestion
    _PROJECT_TASK_LIMIT_MESSAGE = "tasks per project for the organization"
```

File: cvat/settings/base.py [hunk 5]
```diff
@@ -274,6 +278,8 @@ def add_ssh_keys():
     "PROJECT_WEBHOOKS": 10,
 }
 
+ERROR_MESSAGES = {}
```
Comment (nmanovic) [inline cvat/settings/base.py:281]: Please delete

File: tests/python/rest_api/test_limits.py [hunk 6]
```diff
@@ -138,7 +138,7 @@ def test_can_reach_tasks_limit(self):
             self._create_task()
 
         assert capture.value.status == HTTPStatus.FORBIDDEN
-        assert set(json.loads(capture.value.body)) == {self._TASK_LIMIT_MESSAGE}
+        assert -1 < json.loads(capture.value.body)["detail"].rfind(self._TASK_LIMIT_MESSAGE)
```
Comment (nmanovic) [inline tests/python/rest_api/test_limits.py:141]: Will `not  self._TASK_LIMIT_MESSAGE in capture.value.body` work? It looks simpler, isn't it?

File: cvat/apps/iam/rules/limits.rego [hunk 7]
```diff
@@ -27,77 +27,77 @@ check_limit_exceeded(current, max) {
...
-problems contains "org projects limit reached" if {
+problems contains "projects per organization" if {
     check_limit_exceeded(
         input.resource.limits[CAP_ORG_PROJECTS].used,
         input.resource.limits[CAP_ORG_PROJECTS].max
     )
 }
 
-problems contains "org project tasks limit reached" if {
+problems contains "tasks per organization project" if {
```
Comment (kirill-sizov) [inline cvat/apps/iam/rules/limits.rego:65]:
```suggestion
problems contains "tasks per project for organization" if {
```

File: cvat/settings/base.py [hunk 8]
```diff
@@ -192,6 +192,10 @@ def add_ssh_keys():
     'DEFAULT_SCHEMA_CLASS': 'cvat.apps.iam.schema.CustomAutoSchema',
 }
 
+CVAT_ERRORS = {
```
[the remark this answers is not in the document]
Comment (kirill-sizov) [reply cvat/settings/base.py]: I have changed solution

File: cvat-ui/src/reducers/notifications-reducer.ts [hunk 9]
```diff
@@ -721,9 +711,7 @@ export default function (state = defaultState, action: AnyAction): Notifications
                         ...state.messages,
                         models: {
                             ...state.messages.models,
-                            inferenceDone:
-                                'Automatic annotation finished for the ' +
-                                `<a href="/tasks/${taskID}" target="_blank">task ${taskID}</a>`,
+                            inferenceDone: `Automatic annotation accomplished for the [task ${taskID}](/tasks/${taskID})`,
```
Comment (nmanovic) [inline cvat-ui/src/reducers/notifications-reducer.ts:714]:
```suggestion
                            inferenceDone: 'Automatic annotation accomplished for the ' +
                                `[task ${taskID}](/tasks/${taskID})`,
```

**Review by nmanovic — APPROVED** [review]
LGTM

**Comment by kirill-sizov** [comment]
It seems that tests now are fixed

**Pull request merged**